### ISSUE TITLE
Screen PMC for recent publications to index in Wikidata

### ISSUE THREAD

**Issue by Daniel-Mietchen** [Daniel-Mietchen/ideas#684]
One way to get to that is via the PMC API (e.g. https://www.ncbi.nlm.nih.gov/pmc/utils/oa/oa.fcgi?from=2018-01-01+00:00:00 ) but since Fatameh already deals with that, it may be simpler for me to just call Fatameh on a random number in the range of PMCIDs issued so far.

**Comment by Daniel-Mietchen** [comment]
Here's a little shell script that does the job for me: 

```bash
while [ $(date -I) > 0 ]; do pmc=$(python -c 'import random as R; print(R.randint(1, 6000000))'); echo "PMCID: ${pmc}"; curl --header "Authorization: Token 857a8119f0118e1deed6c245ee89548272fd3e2e" https://tools.wmflabs.org/fatameh/token/pmc/add/PMC${pmc}; printf "\n"; done

```

I'm sure this can be made more efficient, and suggestions are welcome. In any case, I'm currently running it as job number 1814.

**Comment by Daniel-Mietchen** [comment]
Obviously, the script may well call Fatameh on papers for which Wikidata items already exist, but due to code improvements, these might actually result in useful edits like https://www.wikidata.org/w/index.php?title=Q39042670&diff=prev&oldid=617550448 .

Likewise, the script does not really look at recent publications, but those are more likely not to be indexed in Wikidata yet. Quite a few older ones are missing too, so checking older PMCIDs occasionally seems worthwhile.

**Comment by Daniel-Mietchen** [comment]
I'm running an additional job 12968 with a more narrow range of values:

```bash
r=5; while [ $r > 0 ]; do pmc=$(python -c 'import random as R; print(R.randint(5500000, 5800000))'); echo "PMCID: ${pmc}"; curl --header "Authorization: Token 857a8119f0118e1deed6c245ee89548272fd3e2e" https://tools.wmflabs.org/fatameh/token/pmc/add/PMC${pmc}; printf "\n"; done
```

**Comment by moqri** [comment]
PMC lists all it's articles in a CSV here which seems to be updated regularly:
ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/PMC-ids.csv.gz

Might be helpful for your screening

**Comment by Daniel-Mietchen** [comment]
I hadn't really thought of using a dump, but this may actually work for the other part of the equation, i.e. getting a list of all PMCIDs (ideally along with their Wikidata IDs). The [query for that](https://query.wikidata.org/#SELECT%20%3Fpmcid%20%3Fitem%20%0AWHERE%0A%7B%0A%20%20%3Fitem%20wdt%3AP932%20%3Fpmcid.%0A%7D%0A) usually times out on the Wikidata Query Service, and I have not been able to get the full list from the [LDF endpoint](https://ldfclient.wmflabs.org/#query=SELECT%20%3Fpmcid%20%3Fitem%20%0AWHERE%0A%7B%0A%20%20%3Fitem%20wdt%3AP932%20%3Fpmcid.%0A%7D%0A) either, though that query does occasionally run through.
